int32 index
---
float32 gain
int32 error